From 757fa354317916763068e656872e16caec72b378 Mon Sep 17 00:00:00 2001
From: Sanyog Kale <skale@habana.ai>
Date: Tue, 30 Nov 2021 12:08:15 +0200
Subject: [PATCH] [SW-0] gaudi2: move arc privileged conf to ppboot

With protection bits configuration done for multiple block in LKD/FW,
it is observed that bootrom flow doesn't restart as expected
upon hard reset. This is due to ARC1 privilege mode configuration
is cleared upon ARCs reset as part of hard reset and ARC1 tries to
access registers of block which are set to privilege or secured by
LKD/firmware.

Apart from bootrom, there are few PLL registers ARC1
accesses in ppboot which are configured as secured/privileged.
This patch moves setting of ARC to privilege as part of ppboot
security init routine (before PCIe link up) and before the
configuration of PLL registers in ppboot.

Change-Id: I37fb262d646dacf741b49e003b43e4c45c84bf54
---

diff --git a/arch/gaudi2/common/security.c b/arch/gaudi2/common/security.c
index 7918a3f..87d9dde 100644
--- a/arch/gaudi2/common/security.c
+++ b/arch/gaudi2/common/security.c
@@ -678,6 +678,8 @@
 
 void ppboot_security_init(void)
 {
+	preboot_secure_arc();
+
 	/* Initialize only if security is enabled */
 	if (!preboot_security_is_enabled())
 		return;
@@ -688,8 +690,6 @@
 
 void preboot_security_init(void)
 {
-	preboot_secure_arc();
-
 	/* Do security related initializations only if security is enabled */
 	if (!preboot_security_is_enabled())
 		return;
